apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -772,18 +776,31 @@ log_tx(#{ tx_hash      := TxHash
         true ->
             St;
         false ->
-            LogEntry = #tx_log_entry{key = Key, value = Info},
+            LogEntry = #tx_log_entry{key = Key,
+                                     value = maps:with([block_hash,
+                                                        block_origin,
+                                                        type], Info)},
             TxLog1 = aesc_window:add(LogEntry, TxLog),
             St#st{tx_log = TxLog1}
     end.
 
-get_latest_tx(Log) ->
-    {#tx_log_entry{key = {_TxHash, _BlockHash},
-                   value = Tx},
+get_latest_tx(Log, C) ->
+    {#tx_log_entry{key = {TxHash, _BlockHash},
+                   value = TxInfo},
      _TxLog1} =
         aesc_window:pop(Log),
-    lager:debug("Tx = ~p", [Tx]),
-    Tx.
+    lager:debug("TxInfo = ~p", [TxInfo]),
+    {TxInfo1, C1} = ensure_signed_tx_included(TxHash, TxInfo, C),
+    {TxHash, TxInfo1, C1}.
+
+ensure_signed_tx_included(TxHash, TxInfo, C) ->
+    case maps:is_key(tx, TxInfo) of
+        true ->
+            {TxInfo, C};
+        false ->
+            {SignedTx, C1} = get_signed_tx(TxHash, C),
```

Comment:
I am trying to understand if `undefined` here is accepted by the code using this function.

Reply:
If there is assumption on fetching tx not returning undefined, I would expect that expectation hardcoded.

Reply:
That would be a very strange occurrence, but the most reasonable action in such a case might be to simply remove that TxHash from the TxLog and pick the next one. But I'm not really sure what exactly it would mean at this point.

Reply:
Would it make sense in that (unlikely? impossible?) case to cause runtime error here, rather than cause invalid value to end up in state and then cause error down the line, maybe in separate event, making it harder to identify cause?

Or at least logging it as warning?

Reply:
or a comment...